src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -9920,101 +10068,681 @@ record A
+            var source =
+@"
+abstract record A
 {
-    public object P { get; }
-    public object P { get; }
-}";
-            var comp = CreateCompilation(src);
-            comp.VerifyDiagnostics(
-                // (4,19): error CS0102: The type 'C' already contains a definition for 'P'
-                //     public object P { get; }
-                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "P").WithArguments("C", "P").WithLocation(4, 19));
-
-            var actualMembers = GetProperties(comp, "C").ToTestDisplayStrings();
-            var expectedMembers = new[]
-            {
-                "System.Type C.EqualityContract { get; }",
-                "System.Object C.Q { get; init; }",
-                "System.Object C.P { get; }",
-                "System.Object C.P { get; }",
-            };
-            AssertEx.Equal(expectedMembers, actualMembers);
-        }
-
-        [WorkItem(44692, "https://github.com/dotnet/roslyn/issues/44692")]
-        [Fact]
-        public void DuplicateProperty_02()
-        {
-            var src =
-@"record C(object P, object Q)
+    internal static bool Report(string s) { System.Console.WriteLine(s); return false; }
+    public abstract bool Equals(A x);
+}
+record B : A
 {
-    public object P { get; }
-    public int P { get; }
-    public int Q { get; }
-    public object Q { get; }
-}";
-            var comp = CreateCompilation(src);
-            comp.VerifyDiagnostics(
-                // (1,27): error CS8866: Record member 'C.Q' must be a readable instance property of type 'object' to match positional parameter 'Q'.
-                // record C(object P, object Q)
-                Diagnostic(ErrorCode.ERR_BadRecordMemberForPositionalParameter, "Q").WithArguments("C.Q", "object", "Q").WithLocation(1, 27),
-                // (4,16): error CS0102: The type 'C' already contains a definition for 'P'
-                //     public int P { get; }
-                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "P").WithArguments("C", "P").WithLocation(4, 16),
-                // (6,19): error CS0102: The type 'C' already contains a definition for 'Q'
-                //     public object Q { get; }
-                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "Q").WithArguments("C", "Q").WithLocation(6, 19));
+    public virtual bool Equals(B other) => Report(""B.Equals(B)"");
+}
+class Program
+{
+    static void Main()
+    {
+        A a1 = new B();
+        A a2 = new B();
 
-            var actualMembers = GetProperties(comp, "C").ToTestDisplayStrings();
-            var expectedMembers = new[]
-            {
-                "System.Type C.EqualityContract { get; }",
-                "System.Object C.P { get; }",
-                "System.Int32 C.P { get; }",
-                "System.Int32 C.Q { get; }",
-                "System.Object C.Q { get; }",
-            };
-            AssertEx.Equal(expectedMembers, actualMembers);
+        System.Console.WriteLine(a1.Equals(a2));
```

Comment:
It may not be the core focus on this PR, but would it be worthwhile exercising `A.Equals(A)` too?

Reply:
> It may not be the core focus on this PR, but would it be worthwhile exercising A.Equals(A) too?

I assume scenario like that has been already covered  and as you mentioned, this is not related to the goal of this PR.

---
In reply to: [451904426](https://github.com/dotnet/roslyn/pull/45647#discussion_r451904426) [](ancestors = 451904426)